### PR TITLE
Backport zipOrAccumulate for Either

### PR DESCRIPTION
**Pull request by nomisRev**

### CODE REVIEW

**Review by serras — APPROVED** [review]
Even with contracts, awesome!

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt [hunk 1]
```diff
@@ -1408,6 +1411,476 @@ public sealed class Either<out A, out B> {
...
+
+
+    public inline fun <E, A, B, Z> zipOrAccumulate(
+      combine: (E, E) -> E,
+      a: Either<E, A>,
+      b: Either<E, B>,
+      transform: (A, B) -> Z,
+    ): Either<E, Z> {
+      contract { callsInPlace(transform, InvocationKind.AT_MOST_ONCE) }
+      return zipOrAccumulate(combine, a, b, unit, unit, unit, unit, unit, unit, unit, unit) { a, bb, _, _, _, _, _, _, _, _ ->
```
Comment (franciscodr) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt:1423]: Should we rename the lambda parameter to `aa` as we do with the rest of the parameters? That would avoid some confusion. Same for the rest of the `zipOrAccumulate` functions below.
```suggestion
      return zipOrAccumulate(combine, a, b, unit, unit, unit, unit, unit, unit, unit, unit) { aa, bb, _, _, _, _, _, _, _, _ ->
```

Comment (nomisRev) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt]: Yes, thank you for catching this @franciscodr 🙏 I updated the code.

**Review by franciscodr — APPROVED** [review]
Thanks, @nomisRev!

**Pull request merged**